src/sidebar/util/version-data.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+/**
+ * An object representing version, browser and user data
+ *
+ * @typedef {Object} VersionData
```

Comment:
`@typedef` is for really defining the shape of objects that don't exist as explicit classes, or you wanted to define an interface that is perhaps a subset of a class. This may be useful for example you have a utility function which is usually passed instances of a particular class but will actually work with any object that has certain properties. If you create a class, the documentation goes above the class definition and documentation for individual properties at the place where they are assigned.

See example at https://stackblitz.com/edit/jsdoc-inference-demo. If you hover references to the class and properties at the bottom of the file, you can see the inferred type and properties.